src/blockdata/block.rs

```diff
@@ -180,13 +237,16 @@ mod tests {
         let real_decode = decode.unwrap();
         assert_eq!(real_decode.header.version, 1);
         assert_eq!(serialize(&real_decode.header.prev_blockhash), prevhash);
-        // [test] TODO: actually compute the merkle root
+        assert_eq!(real_decode.header.merkle_root, real_decode.merkle_root());
         assert_eq!(serialize(&real_decode.header.merkle_root), merkle);
         assert_eq!(real_decode.header.time, 1231965655);
         assert_eq!(real_decode.header.bits, 486604799);
         assert_eq!(real_decode.header.nonce, 2067413810);
         // [test] TODO: check the transaction data
-    
+
+        // should be also ok for a non-witness block as commitment is optional in that case
+        assert!(real_decode.check_witness_commitment());
```

Comment:
Would be nice to check the inverse case (ie add a random witness to a tx and check that it fails) too.